src/utils.js

```diff
@@ -589,30 +588,24 @@ async function imageminGenerate(original, minimizerOptions) {
     return original;
   }
 
-  const { ext: extOutput } = fileTypeFromBuffer(result) || {};
-  const extInput = path.extname(original.filename).slice(1).toLowerCase();
+  const {
+    dir: fileDir,
+    name: fileName,
+    ext: fileExt,
+  } = path.parse(original.filename);
 
-  let newFilename = original.filename;
-
-  if (extOutput && extInput !== extOutput) {
-    newFilename = original.filename.replace(
```

Comment:
It can be URL, i.e. with `?test=1` due legacy reasons, better to keep it as is

Reply:
but i have already used this way of generating filename in implementation and its passed all tests
https://github.com/webpack-contrib/image-minimizer-webpack-plugin/blob/52966fe1a6d2125befa22152be347b1b53b78130/src/utils.js#L1080

Reply:
hm, sounds like we don't have legacy tests, but it is possible, but we have wrong logic too, because we use `$`, let's return this as is, I will look at this deeply later (need check it inside webpack)

Reply:
ok, I reverted those changes